nemo/collections/nlp/models/nlp_model.py

```diff
@@ -410,3 +522,160 @@ def save_checkpoint(self, filepath, weights_only: bool):
                     )
                     atomic_save(checkpoint, filepath)
         return None
+
+
+class NLPDDPPlugin(DDPPlugin):
```

Comment:
Consider moving this out of nlp_model.py into ints own file.

Reply:
I think that makes sense. I'll have to figure out how to organize it.

Reply:
Done. Also moved NLPCheckpointConnector.